todocli/todo_api/queries.py

```diff
@@ -0,0 +1,76 @@
+from typing import Union
+
+
+from todocli.rest_request import RestRequestGet
+from todocli.todo_api import api_urls
+from todocli.todo_api.exceptions import (
+    TaskNotFoundByName,
+    TaskNotFoundByIndex,
+    ListNotFound,
+)
+
+list_ids_cached = {}
+
+
+def query_tasks(list_name: str, include_completed=False, num_tasks: int = 100):
+    query_url = api_urls.get_tasks(
+        get_list_id_by_name(list_name), num_tasks, include_completed=include_completed
+    )
+
+    return RestRequestGet(query_url).execute()
+
+
+def query_task(list_name: str, task_name: str):
+    query_url = api_urls.get_task_by_name(get_list_id_by_name(list_name), task_name)
+    try:
+        return RestRequestGet(query_url).execute()[0]
+    except IndexError:
+        raise TaskNotFoundByName(task_name, list_name)
+
+
+def query_lists():
+    return RestRequestGet(api_urls.get_all_lists()).execute()
+
+
+def get_task_id_by_name(list_name: str, task_name: str):
+    result_task = query_task(list_name, task_name)
+    return result_task["id"]
+
+
+def get_task_id_by_list_position(list_name: str, task_list_position):
+    tasks = query_tasks(list_name, task_list_position + 1)
```

Comment:
Does this work? `query_tasks` has arguments `str, bool (optional), int (optional)`.

Reply:
Also, why is it `+1`?